registry/extension/oras/referrers.go

```diff
@@ -41,7 +69,42 @@ func (h *referrersHandler) getReferrers(w http.ResponseWriter, r *http.Request)
 		referrers = []orasartifacts.Descriptor{}
 	}
 
-	response := referrersResponse{
+	// only consider pagination if # of referrers is greater than page size
+	if len(referrers) > nPage {
+		startIndex := 0
+		if len(nextTokenMap) > 0 {
+			for i, ref := range referrers {
+				// check if ref matches a digest in nextToken list
+				if _, ok := nextTokenMap[ref.Digest.String()]; ok {
+					// set the starting index to the largest index
+					if (i + 1) > startIndex {
+						startIndex = i + 1
+					}
+				}
+			}
+		}
+
+		// only applicable if last item provided as nextToken
+		if startIndex == len(referrers) {
+			referrers = []orasartifacts.Descriptor{}
+		}
+
+		// if there's only 1 page of results left
+		if len(referrers)-startIndex <= nPage {
+			referrers = referrers[startIndex:]
+		} else {
+			referrers = referrers[startIndex:(startIndex + nPage)]
+			// generate string list of digests for nextToken
+			var nextDgsts []string
+			for i := nPage - 1; i >= nPage-minPageSize; i-- {
+				nextDgsts = append(nextDgsts, referrers[i].Digest.String())
+			}
+			// add the Link Header
+			w.Header().Set("Link", generateLinkHeader(h.extContext.Repository.Named().Name(), h.Digest.String(), artifactType, nextDgsts, nPage))
```

Comment:
Should be base64 encoded to avoid conflicts. (https://www.ietf.org/rfc/rfc2396.txt, section 2.2)

Reply:
base64 encoded the entire Link header

Reply:
If the entire link header is encoded, it would be harder to read it for somebody to diagnose issues. Suggest we only encode the nextToken part since the other parts are URL safe

Reply:
switched to just encoding the nextToken